packages/core/src/generatePolicy.js

```diff
@@ -97,7 +97,7 @@ function createModuleInspector (opts = {}) {
     // skip json files
     const filename = moduleRecord.file || 'unknown'
     const fileExtension = path.extname(filename)
-    if (fileExtension !== '.js') {
+    if (fileExtension === '.json') {
```

Comment:
I don't have a strong opinion on that. But @naugtur recommended that we just invert to an equality check. Either way is fine by me.

Reply:
I think we should avoid JSON as described in the comment above and allow everything else by default if it's imported in your bundle

Then we could make sure errors in parsing are only logged as warnings for extensions different than the list of known and expected javascript extensions. I wouldn't put ts there tho. So if someone misconfigures stuff, we'd throw, but only if ts files are actual typescript, not extension left on a processed file.

@legobeat does this sound right?

Reply:
From a snaps point of view, the problem is that Browserify/babelify doesn't change the file extension of transpiled files. So this check fails. If we go with an allowlist approach that doesn't include `.ts` we will also have the same problem long-term in the MetaMask extension with more and more code being migrated to TypeScript.

I'm agnostic whether we choose the allowlist approach or simply ignore JSON. But we need a solution for parsing the Browserify `.ts` output regardless.

This could perhaps be handled by the LavaMoat browserify plugin before generating the policy, but that seems even more complicated.